src/java.base/share/classes/sun/security/ssl/CompressionAlgorithm.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2022 THL A29 Limited, a Tencent company. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package sun.security.ssl;
+
+import java.util.AbstractMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Enum for (D)TLS certificate compression algorithms.
+ */
+enum CompressionAlgorithm {
+    ZLIB                (1,  "zlib"),
+    BROTLI              (2,  "brotli"),
+    ZSTD                (3,  "zstd");
+
+    final int id;
+    final String name;
+
+    CompressionAlgorithm(int id, String name) {
+        this.id = id;
+        this.name = name;
+    }
+
+    static CompressionAlgorithm nameOf(String name) {
```

Comment:
Shouldn't this be valueOf?  We're not consistent in the method name across all our sun.security.ssl enumerations, but more often than not when we return the enumeration object it's via a valueOf call.

Reply:
Hmmm...I take that back.  Seems like when the input is a string we do go by nameOf.  Maybe this is fine.